.github/workflows/kamaldeploy.yml

```diff
@@ -20,34 +19,26 @@ jobs:
           ref: main
 
       - name: Get the latest commit SHA
-        id: commit_sha
         run: echo "SHA=$(git rev-parse HEAD)" >> $GITHUB_ENV
 
-      - name: Check commit status (checks for passed tests)
-        id: commit_status
+      - name: Confirm all checks passed
         run: |
-          COMMIT_SHA="${{ env.SHA }}"
-          echo "Checking status of commit ${COMMIT_SHA}..."
-
-          # Get the status of the latest commit using GitHub API
+          # Returns json data on all checks that were run
           response=$(curl -L \
           -H "Accept: application/vnd.github+json" \
-          -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
-          -H "X-GitHub-Api-Version: 2022-11-28" \
-          "https://api.github.com/repos/${{ github.repository }}/commits/${COMMIT_SHA}/status")
+          "https://api.github.com/repos/${{ github.repository }}/commits/${{ env.SHA }}/check-runs")
 
-          if echo "$response" | grep -q '"state": "success"'; then
-            echo "commit_status=success" >> $GITHUB_ENV
-          else
-            echo "Commit ${COMMIT_SHA} has failed checks."
-            echo "commit_status=failure" >> $GITHUB_ENV
-            exit 1
-          fi
+          # Check the conclusion of eash check for pass/fail
+          for conclusion in $(echo "$response" | jq -r '.check_runs[].conclusion'); do
+            if [ "$conclusion" != "success" ]; then
+              echo "Commit "${{ env.SHA }}" has failed checks."
+              exit 1
+            fi
+          done
 
   deploy:
     runs-on: ubuntu-latest
     needs: check_latest_commit
-    if: needs.check_latest_commit.outputs.commit_status == 'success'
 
```

Comment:
Since we exit with code 1 (failure/error) if there are any fail check, we don't need to check for passing here. This job with never run as it needs "check_latest_commit" job to be be completed without error.